src/combinatorics/simple_graph/strongly_regular.lean

```diff
@@ -0,0 +1,147 @@
+/-
+Copyright (c) 2021 Alena Gusakov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Author: Alena Gusakov
+-/
+import combinatorics.simple_graph.basic
+/-!
+# Strongly regular graphs
+
+## Main definitions
+
+* `common_neighbors` is the intersection of the neighbor sets of two given vertices
+
+* `G.is_SRG_of n k l m` is a structure for a `simple_graph` with conditions
+  * The cardinality of the vertex set is `n`
+  * `G` is a regular graph with degree `k`
+  * The number of common neighbors between any two adjacent vertices in `G` is `l`
+  * The number of common neighbors between any two nonadjacent vertices in `G` is `m`
+
+## TODO
+- Define the complement of a graph
+- Prove that the complement of a strongly regular graph is strongly regular
+- Prove that the parameters of a strongly regular graph
+  obey relation `(n - k - 1) * m = k * (k - l - 1)`
+- Prove that if I is the identity matrix and J is the all-one matrix,
+  then the adj matrix A of SRG obeys relation A^2 = kI + lA + m(J - I - A)
+-/
+universes u
+
+namespace simple_graph
+variables {V : Type u}
+variables (G : simple_graph V)
+
+variables  [fintype V] [locally_finite G]
```

Comment:
You can safely remove `[locally_finite G]` since it's automatically implied from `[fintype V]`.